### PR TITLE
DDCE-518 CharityScottishRegulatorRegistrationNumber page

### PR DESCRIPTION
**Pull request by prameya-phatak**
# DDCE-518

##New feature

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: conf/messages.cy [hunk 1]
```diff
@@ -278,3 +278,13 @@ charityCommissionRegistrationNumber.error.format = Registration number must be 6
...
+
+# CharityScottishRegulatorRegistrationNumberPage Messages
+# ----------------------------------------------------------
+charityScottishRegulatorRegistrationNumber.title = What is your Office of the Scottish Charity Regulator registration number?
+charityScottishRegulatorRegistrationNumber.heading = What is your Office of the Scottish Charity Regulator registration number?
+charityScottishRegulatorRegistrationNumber.error.required = Enter your charity''s registration number
+charityScottishRegulatorRegistrationNumber.error.format = Registration number is SC then 0 and 5 numbers, like SC012345
+charityScottishRegulatorRegistrationNumber.hint = For example, SC012345
+charityScottishRegulatorRegistrationNumber.help = Where to find the registration number?
+charityScottishRegulatorRegistrationNumber.hintText = The registration number will be on any mail the regulator has sent you.
```
Comment (anburoach) [inline conf/messages.cy:290]: Add new line at the end

File: test/controllers/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.regulatorsAndDocuments
+
+import base.SpecBase
+import controllers.actions.{AuthIdentifierAction, FakeAuthIdentifierAction}
+import forms.regulatorsAndDocuments.{CharityCommissionRegistrationNumberFormProvider, CharityScottishRegulatorRegistrationNumberFormProvider}
```
Comment (anburoach) [inline test/controllers/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberControllerSpec.scala:21]: Remove unused imports

File: app/controllers/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberController.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import navigation.RegulatorsAndDocumentsNavigator
+import pages.Section2Page
+import pages.regulatorsAndDocuments.CharityScottishRegulatorRegistrationNumberPage
+import play.api.mvc._
+import repositories.UserAnswerRepository
+import views.html.regulatorsAndDocuments.CharityScottishRegulatorRegistrationNumberView
+
+import scala.concurrent.Future
+
+class CharityScottishRegulatorRegistrationNumberController @Inject()(
```
Comment (abhijeetardale) [inline app/controllers/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberController.scala:34]: File name is too long, can we trim like ScottishRegulatorRegNumber and it applies to all files

File: app/navigation/RegulatorsAndDocumentsNavigator.scala [hunk 4]
```diff
@@ -39,10 +40,15 @@ class RegulatorsAndDocumentsNavigator @Inject()(implicit frontendAppConfig: Fron
           case _ => routes.SessionExpiredController.onPageLoad()
         }
 
-        case CharityCommissionRegistrationNumberPage => userAnswers: UserAnswers => userAnswers.get(CharityCommissionRegistrationNumberPage) match {
-          case Some(_) => routes.IndexController.onPageLoad()        // TODO modify once next page created
-          case _ => routes.SessionExpiredController.onPageLoad()     // TODO modify once next page created
-        }
+    case CharityCommissionRegistrationNumberPage => userAnswers: UserAnswers => userAnswers.get(CharityCommissionRegistrationNumberPage) match {
+      case Some(_) => regulatorAndDocumentInfoRoutes.CharityScottishRegulatorRegistrationNumberController.onPageLoad(NormalMode)
```
Comment (abhijeetardale) [inline app/navigation/RegulatorsAndDocumentsNavigator.scala:44]: could you please check exact match with Page Id, as we discussed in call, 
case Some(items)  if items.toSeq.contains(ScottishRegulatorRegistrationNumber) =>

File: app/views/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberView.scala.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+@import views.ViewUtils._
+@import config.FrontendAppConfig
+
+@this(
+    govukLayout: templates.GovukLayoutWrapper,
+    formHelper: FormWithCSRF,
+    errorSummary: components.errorSummary,
+    button: components.button,
+    inputText: components.inputText,
+    h1: components.h1,
```
Comment (anburoach) [inline app/views/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberView.scala.html:28]: h1 and subheading are not required so remove it

File: app/navigation/RegulatorsAndDocumentsNavigator.scala [hunk 6]
```diff
@@ -39,10 +40,15 @@ class RegulatorsAndDocumentsNavigator @Inject()(implicit frontendAppConfig: Fron
...
+      case _ => routes.SessionExpiredController.onPageLoad()
+    }
+
+    case CharityScottishRegulatorRegistrationNumberPage => userAnswers: UserAnswers => userAnswers.get(CharityScottishRegulatorRegistrationNumberPage) match {
+      case Some(_) => routes.IndexController.onPageLoad()        // TODO modify once next page created
```
Comment (abhijeetardale) [inline app/navigation/RegulatorsAndDocumentsNavigator.scala:49]: same as above

File: app/navigation/RegulatorsAndDocumentsNavigator.scala (hunk 4 above)
Comment (prameya-phatak) [reply app/navigation/RegulatorsAndDocumentsNavigator.scala:44]: there is separate ticket for this change.so will cover this in that ticket

File: app/controllers/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberController.scala (hunk 3 above)
Comment (prameya-phatak) [reply app/controllers/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberController.scala:34]: changed the file name

File: app/views/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberView.scala.html (hunk 5 above)
Comment (prameya-phatak) [reply app/views/regulatorsAndDocuments/CharityScottishRegulatorRegistrationNumberView.scala.html:28]: removed

File: app/navigation/RegulatorsAndDocumentsNavigator.scala [hunk 7]
```diff
@@ -22,8 +22,9 @@ import controllers.routes
 import javax.inject.Inject
 import models._
 import pages.Page
-import pages.regulatorsAndDocuments.{CharityCommissionRegistrationNumberPage, CharityRegulatorPage, IsCharityRegulatorPage}
+import pages.regulatorsAndDocuments.{CharityCommissionRegistrationNumberPage, ScottishRegulatorRegNumberPage, IsCharityRegulatorPage,CharityRegulatorPage}
 import play.api.mvc.Call
+import controllers.regulatorsAndDocuments.{routes =>regulatorAndDocumentInfoRoutes}
```
Comment (anburoach) [inline app/navigation/RegulatorsAndDocumentsNavigator.scala:27]: Remove unused import

**Pull request merged**